src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -161,7 +174,25 @@ public Object newInstance(Object ctrParamObj)
                     }
                 } else  if (type.equals("KeyPairGenerator")) {
                     if (algo.equals("EC")) {
-                        return new ECKeyPairGenerator();
+                        /* Disabling OpenSSL usage in AIX due to perfomance regression observed. */
+                        if (useNativeECKeyGen &&
+                           (NativeCrypto.getVersionIfAvailable() >= NativeCrypto.OPENSSL_VERSION_1_1_0) &&
+                           (!System.getProperty("os.name").contains("AIX"))) {
+                            if (nativeCryptTrace) {
+                                System.err.println("EC KeyPair Generation - Using OpenSSL integration.");
+                            }
+                            return new NativeECKeyPairGenerator();
+                        } else {
```

Comment:
This yields misleading trace output if `useNativeECKeyGen == false`; suggest:
```java
                        if (useNativeECKeyGen) {
                            if (NativeCrypto.getVersionIfAvailable() < NativeCrypto.OPENSSL_VERSION_1_1_0) {
                                if (nativeCryptTrace) {
                                    System.err.println("EC KeyPair Generation - Not using OpenSSL integration due to older version of OpenSSL (<1.1.0).");
                                }
                            } else if (isAIX) {
                                /* Disabling OpenSSL usage on AIX due to perfomance regression observed. */
                                if (nativeCryptTrace) {
                                    System.err.println("EC KeyPair Generation - Not using OpenSSL integration on AIX.");
                                }
                            } else {
                                if (nativeCryptTrace) {
                                    System.err.println("EC KeyPair Generation - Using OpenSSL integration.");
                                }
                                return new NativeECKeyPairGenerator();
                            }
                        }
                        return new ECKeyPairGenerator();
```

Reply:
Changed to suggested.